### PR TITLE
Adding modeller functionality for mutations

### PR DESCRIPTION
**Pull request by steven-albanese**
Still need to add a modeller test

### CODE REVIEW

**Comment by steven-albanese** [comment]
Just realized I need to add `modeller` and the license in order for the tests to work

**Comment by andrrizzi** [comment]
> Just realized I need to add modeller and the license in order for the tests to work

Ah, right! I'm not sure if we can do something similar to what we do with the openeye license.

**Comment by steven-albanese** [comment]
From the modeller website: 
```
You will be prompted after installation to edit a file to add your Modeller license key. Alternatively, set the KEY_MODELLER environment variable to your license key before you run 'conda install'.
```

File: Yank/pipeline.py [hunk 1]
```diff
@@ -943,6 +945,111 @@ def add_missing_atoms(value):
...
+            model.build(initialize_xyz=False, build_method='INTERNAL_COORDINATES')
+
+        else:
+            logger.info('pdbfixer: No mutations will be applied since "WT" specified.')
+
+    process_directive('apply_mutations', apply_mutations_modeller, None)
+
+    # Check that there were no extra options
+    if len(directives) > 0:
+        raise ValueError("The 'pdbfixer:' block contained some nodes that it didn't know how to process: {}".format(directives))
```
[the remark this answers is not in the document]
Comment (steven-albanese) [reply Yank/pipeline.py]: Thanks for catching this!

**Comment by steven-albanese** [comment]
@jchodera @Lnaden do you have any thoughts on how to handle the license? It's a string of letters that seems to be constant for the last few years, but I'm not sure if we need to protect it somehow for travis

**Comment by Lnaden** [comment]
https://docs.travis-ci.com/user/environment-variables/#defining-encrypted-variables-in-travisyml

**Comment by steven-albanese** [comment]
Ok it looks like the builds are passing now. The only outstanding issue is whether I should check to make sure that no double mutations are being passed

**Comment by Lnaden** [comment]
That is probably worth testing, since our pdbfixer supports double mutants

**Comment by steven-albanese** [comment]
Ok -- I'll raise a `ValueError` if a double mutant gets passed that prints the offending mutant and says it's not supported by modeller right now

**Comment by steven-albanese** [comment]
I think this is ready when someone has a chance to look at it! Let me know if there are any other edits I should make.

File: .travis.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ install:
   - source devtools/travis-ci/install.sh
   - export PYTHONUNBUFFERED=true
   # Add org channel
-  - conda config --add channels omnia --add channels conda-forge
+  - conda config --add channels omnia --add channels conda-forge --add channels salilab
```
Comment (andrrizzi) [inline .travis.yml]: It looks like the `omnia` channel already has a `modeller` package. Not sure if it's the same.

Comment (steven-albanese) [reply .travis.yml]: I'm not sure, but I *think* they are different

**Comment by steven-albanese** [comment]
Updated the docs and also fixed the default behavior for when `chain_id` isn't selected

File: docs/yamlpages/molecules.rst [hunk 3]
```diff
@@ -239,6 +239,58 @@ PDBFixer is applied after ``strip_protons`` if both are requested.
...
+* ``T315I`` : a single mutation that changes Thr at resid 315 to Ile
+
+``chain_id`` can be either an integer or string. Strings are case sensitive!
+If ``chain_id`` is not specified, it defaults to ``0`` (the first chain in the file will be selected).
+
+Examples for specifying ``chain_id``:
+* ``0`` : the first chain in the file will be selected
+* ``B`` : chain B will be selected
+
+Modeller is applied after both ``strip_protons`` and ''pdbfixer'' if either (or both) are requested.
```
Comment (andrrizzi) [inline docs/yamlpages/molecules.rst]: I think you meant to use \` instead of ' for `pdbfixer` here.

**Pull request merged**